File: app/src/main/java/com/platdmit/simplecloudmanager/data/database/dao/BaseDao.java
package com.platdmit.simplecloudmanager.data.database.dao;

import java.util.List;

import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Update;

public interface BaseDao<Db> {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insert(Db dbElement);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertList(List<Db> dbElements);

    @Update
    void update(Db dbElement);

    @Update
    void updateAll(List<Db> dbElements);

    @Delete
    void delete(Db dbElement);
}

File: data/src/main/java/com/platdmit/data/database/dao/BaseDao.kt
package com.platdmit.data.database.dao

import androidx.room.Delete
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Update

interface BaseDao<Db> {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insert(dbElement: Db)

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertList(dbElements: List<Db>)

    @Update
    fun update(dbElement: Db)

    @Update
    fun updateAll(dbElements: List<Db>)

    @Delete
    fun delete(dbElement: Db)
}